x-pack/plugins/maps/public/classes/styles/vector/components/legend/vector_style_legend.tsx

```diff
@@ -62,7 +61,8 @@ export function VectorStyleLegend({
     );
   }
 
-  function renderMasksByFieldOrigin(fieldOrigin: FIELD_ORIGIN) {
+  function MasksByFieldOrigin({ fieldOrigin }: { fieldOrigin: FIELD_ORIGIN }) {
```

Comment:
Now that `MasksByFieldOrigin` is a react component, how about moving it into a separate file?